oap-stdlib/src/main/java/oap/message/MessageSender.java

```diff
@@ -532,10 +550,21 @@ public MessageSender syncDisk() {
 
     private boolean isValidMessage( Path messagePath ) {
         try {
-            return !java.nio.file.Files.isDirectory( messagePath )
-                && ( messagePath.toString().endsWith( ".bin" ) || messagePath.toString().endsWith( ".lock" ) )
-                && ByteSequence.of( Hex.decodeHex( FilenameUtils.removeExtension( FilenameUtils.getName( messagePath.toString() ) ) ) ) != null;
-        } catch( DecoderException e ) {
+            if( java.nio.file.Files.isDirectory( messagePath ) ) return false;
+            String messageFile = messagePath.toString();
+            if( !messageFile.endsWith( ".bin" ) && !messageFile.endsWith( ".lock" ) ) return false;
+
+            var md5HexWithVersion = FilenameUtils.removeExtension( FilenameUtils.getName( messagePath.toString() ) );
+            var versionStart = md5HexWithVersion.lastIndexOf( '-' );
+            if( versionStart > 0 ) {
+                Short.parseShort( md5HexWithVersion.substring( versionStart + 1 ) );
+                ByteSequence.of( Hex.decodeHex( md5HexWithVersion.substring( 0, versionStart ) ) );
+            } else {
+                ByteSequence.of( Hex.decodeHex( md5HexWithVersion ) );
+            }
+
+            return true;
+        } catch( DecoderException | NumberFormatException e ) {
```

Comment:
`log.debug` or even `log.error`. `(The message version {} is not valid, versionStart)`

Reply:
Got the explanation offline - if the message `isValid` = false the exception will be thrown further.